File: Airline Reservation System/backend/src/main/java/com/coforge/ars/service/WalletService.java
package com.coforge.ars.service;

import java.math.BigDecimal;
import java.util.List;

import java.util.Optional;

import com.coforge.ars.model.Wallet;


public interface WalletService {
	boolean saveWallet(Wallet wallet);
	boolean updateWallet(Integer walletId,Wallet wallet);
	boolean deleteWalletById(Integer WalletId);
	Optional<Wallet> findByWalletId(Integer walletId);
	List<Wallet> findAllWallets();
	
	boolean addMoney(Integer customerId, Double amount);
	BigDecimal checkBalance(Integer customerId);

}

File: Airline Reservation System/frontend/src/services/WalletService.js
import axios from "axios";

const BASE_URL = "http://localhost:1515/api/v1/customer/wallets";

class WalletService {

    saveWallet(wallet) {
        return axios.post(BASE_URL, wallet);
    }

    findAllWallets() {
        return axios.get(BASE_URL);
    }

    findWalletById(id) {
        return axios.get(BASE_URL + "/" + id);
    }

    updateWallet(id, wallet) {
        return axios.put(BASE_URL + "/" + id, wallet);
    }

    deleteWallet(id) {
        return axios.delete(BASE_URL + "/" + id);
    }

    addMoney(customerId, amount) {
        return axios.put(
            "http://localhost:1515/api/v1/customer/wallets/addmoney/" +
            customerId + "/" + amount
        );
    }

    checkBalance(customerId) {
        return axios.get(
            "http://localhost:1515/api/v1/customer/wallets/balance/" +
            customerId
        );
    }

}

export default new WalletService();